src/prompt.c

```diff
@@ -959,10 +959,14 @@ run_prompt_command(struct view *view, const char *argv[])
 		}
 
 	} else if (!strcmp(cmd, "goto")) {
-		if (!argv[1] || !strlen(argv[1]))
+		if (!argv[1] || !strlen(argv[1])) {
 			report("goto requires an argument");
-		else
-			goto_id(view, argv[1], true, true);
+		} else {
+			char dest[SIZEOF_STR];
+
+			snprintf(dest, SIZEOF_STR, "%s^{}", argv[1]);
+			goto_id(view, dest, true, true);
```

Comment:
LGTM

Reply:
Thanks for the review Johannes. As one can also just do `:<sha1>` at the prompt, I'll amend the PR to move the code directly in goto_id().